### PR TITLE
Add missing rgw configs

### PR DESCRIPTION
**Pull request by hemanthnakkina**
Add RGW configs rgw_swift_account_in_url,
rgw_keystone_implicit_tenants when
namespace-projects is True.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Unit test added.
Manual testing

## Contributor's Checklist

Please check that you have:

- [x] self-reviewed the code in this PR.
- [ ] added code comments, particularly in hard-to-understand areas.
- [ ] updated the user documentation with corresponding changes.
- [x] added tests to verify effectiveness of this change.

### CODE REVIEW

File: tests/unit/test_charm.py [hunk 1]
```diff
@@ -175,6 +175,52 @@ def test_all_relations_with_enable_rgw_config(self, mock_file, subprocess, cclie
...
+                "enable",
+                "rgw",
+            ],
+            capture_output=True,
+            text=True,
+            check=True,
+            timeout=180,
+        )
+        # Verify if update_config for RGW configs is called
+        self.assertEqual(cclient.from_socket().cluster.update_config.call_count, 15)
```
Comment (UtkarshBhatthere) [inline tests/unit/test_charm.py:222]: Is there any specific reason for the numerical 15 call count ?

File: src/charm.py [hunk 2]
```diff
@@ -513,6 +515,14 @@ def configure_rgw_service(self, event: ops.framework.EventBase) -> None:
                     "rgw_keystone_service_token_accepted_roles": self.id_svc.interface.admin_role,
                     "rgw_s3_auth_use_keystone": str(True).lower(),
                 }
+                namespace_projects = self.leader_get("namespace-projects")
```
Comment (UtkarshBhatthere) [inline src/charm.py:518]: When does the corresponding `set` operation for `namespace-projects` take place ? As in, who sets this key value.

File: tests/unit/test_charm.py (hunk 1 above)
Comment (hemanthnakkina) [reply tests/unit/test_charm.py:222]: Based on number of rgw configs that are set.

**Review by UtkarshBhatthere — COMMENTED** [review]
Some questions

File: src/charm.py (hunk 2 above)
Comment (hemanthnakkina) [reply src/charm.py:518]: https://github.com/canonical/charm-microceph/blob/main/src/charm.py#L385. The value will be set during deploy time only.

**Comment by hemanthnakkina** [comment]
> Some questions

Replied inline

File: tests/unit/test_charm.py (hunk 1 above)
Comment (UtkarshBhatthere) [reply tests/unit/test_charm.py:222]: Ack, every time the config key list is changes the devs will have to take a look at this test case (which is good in my opinion). Can we change the numerical to a script level const variable though ? I would prefer to have a tidy set of constants defined (with a comment explaining their value).

File: src/charm.py (hunk 2 above)
Comment (UtkarshBhatthere) [reply src/charm.py:518]: perfect. This value won't change throughout the charm's life.

File: tests/unit/test_charm.py (hunk 1 above)
Comment (hemanthnakkina) [reply tests/unit/test_charm.py]: Done

**Review by UtkarshBhatthere — APPROVED** [review]
LGTM

**Comment by UtkarshBhatthere** [comment]
waiting for the CI to be green.

**Comment by hemanthnakkina** [comment]
I tested the Juju upgrade test in my local and its successful. I am re-triggering the build for 3rd time. Seems like intermittent issues.

**Pull request merged**